File: Models/Spot.cs
using System.Collections.Generic;

public class Spot
{
    public string Name { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public List<Direction> Directions { get; set; }

    public Spot(string name, double latitude, double longitude, List<Direction> directions)
    {
        Name = name;
        Latitude = latitude;
        Longitude = longitude;
        Directions = directions;
    }
}

File: src/models/spot.rs
use super::date::DateSpan;

pub struct Spot {
    pub name: String,
    pub latitude: f32,
    pub longitude: f32,
    pub directions: Vec<Direction>,
    pub inactive_dates: Vec<DateSpan>
}

pub struct Direction {
    pub minimum: u32,
    pub maximum: u32
}